analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,394 @@
+
+```{r}
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 90% of the total counts are
+# for that particular status
+cytoband_status_count <- consensus_seg_cytoband_status_df %>%
+  # Filter the annotated CN data to include only neutral chromosome arms
+  filter(chromosome_arm %in% neutral_arms) %>%
+  select(
+    Kids_First_Biospecimen_ID,
+    cytoband,
+    loss_fraction,
+    gain_fraction,
+    callable_fraction,
+    dominant_status = dominant_cytoband_status
+  ) %>%
+  mutate(
+    dominant_status = case_when(
+      callable_fraction < (1 - uncallable_threshold) ~ "uncallable",
+      loss_fraction > status_threshold ~ "loss",
+      gain_fraction > status_threshold ~ "gain",
+      loss_fraction + gain_fraction > status_threshold ~ "unstable",
+      TRUE ~ "neutral"
+    )
+  )
+
+# Display table
+cytoband_status_count %>%
+  group_by(dominant_status)
+```
+
+### Determine gene-level status
+
+```{r}
+# These are the cytobands that have not been defined as gain or loss --
+# we want to define their gene-level status
+neutral_cytobands <- cytoband_status_count %>%
+  filter(dominant_status %in% c("unstable", "neutral")) %>%
+  dplyr::pull("cytoband")
+
+# Filter the annotated CN data to include only these cytobands
+neutral_status_cytoband_df <- consensus_seg_cytoband_status_df %>%
+  filter(cytoband %in% neutral_cytobands)
+
+# Join gene-level data
+combined_neutral_status_cytoband_df <- neutral_status_cytoband_df %>%
+  left_join(
+    consensus_seg_autosomes_df,
+    by = c("Kids_First_Biospecimen_ID" = "biospecimen_id", "cytoband")
+  )
+```
+
+```{r}
+if (!(running_in_ci)) {
+  # Now count the gene-level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(
+    combined_neutral_status_cytoband_df,
+    "gene_symbol",
+    status_threshold
+  )
+
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the fraction of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the chromosome arm calls
+# plot_dominant_status_calls(
+#   arm_status_count
+# )
+```
+
+### Plot cytoband status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the cytoband calls
+# plot_dominant_status_calls(
+#   cytoband_status_count
+# )
+```
+
+### Plot gene status calls
+
+```{r}
+# if (!(running_in_ci)) {
+#   # Run `plot_dominant_status_calls` function for the gene-level calls
+#   plot_dominant_status_calls(gene_status_count)
+# }
+```
+
+## Combine arm, cytoband, and gene-level status data
+
+```{r}
+# For each of the datasets we're joining, we'll only keep these columns:
+cols_to_keep <- c("Kids_First_Biospecimen_ID", "dominant_status")
```

Comment:
If we do leave the upper dfs with `dominant_arm_status` and `dominant_cytoband_status`, this may need to change. I might have a small preference for leaving the more specific column names and changing only down here, but that is a small preference, so either way you feel comfortable.